### DISCUSSION TITLE
How to handle HMR rehydration in development server?

### DISCUSSION THREAD

**Discussion by jcayzac** [frejs/fre#241] [Q&A]
I'm using fre with vitejs. I get browser events whenever I change a file, but I don't know what to call in fre to rehydrate. Should I just call `render()` again? Won't that lose state, tho?

**Comment by yisar** [comment]
In fact, this is more difficult, I have tried, you need to ast analysis of the changed file, and then find the corresponding component, and then call scheduleUpdate (theComponent)

https://github.com/yisar/deku/blob/master/src/client.ts

Are you interested in this?  I am willing to assist you to complete a prototype implement.

**Reply by jcayzac** [reply]
It seems very close to what I've seen some vite plugins do. I'll try to make it work when I find the time. Thanks!

**Reply by yisar** [reply]
Can you make the repo publicly? I can assist you and pull prs for you.

Thank you.

**Reply by jcayzac** [reply]
I've been working on this branch: https://github.com/jcayzac/vite-react-ts-eslint-prettier-tailwindcss/tree/feature/fre

From the Vite docs, it seems all is needed is a plugin with a [`handleHotUpdate`](https://vitejs.dev/guide/api-plugin.html#handlehotupdate) hook. The [vue plugin](https://github.com/vitejs/vite/blob/main/packages/plugin-vue/src/handleHotUpdate.ts) has code that's similar to what you're doing in Deku.
